themes/classic/templates/_partials/microdata-jsonld.tpl

```diff
@@ -0,0 +1,201 @@
+{**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ *}
+ 
+{*Let the meta managed by the product.tpl for specific product meta*}
+ {if !isset($product) && $page.page_name != 'product'}
+  <meta property="og:title" content="{$page.meta.title}" />
+  <meta property="og:description" content="{$page.meta.description}" />
+  <meta property="og:type" content="website" />
+  <meta property="og:url" content="{$urls.current_url}" />
+  <meta property="og:site_name" content="{$shop.name}" />
```

Comment:
We can add image here 😎


```
{if isset($product) && isset($product.cover) && $page.page_name == 'product'}
    <meta name="thumbnail" content="{$product.cover.bySize.large_default.url}" />
    <meta property="og:image" content="{$product.cover.bySize.large_default.url}"/>
{else}
    <meta property="og:image" content="{$urls.shop_domain_url}{$shop.logo}"/>
{/if}
```

Reply:
I let that in the product.tpl page... (meta tags in html flow, microdata in json-ld)... I only put the meta in JSON-LD file .tpl for the pages that are not "product" pages because the tags are missing.
But yes it can be a discussion : centralize the meta tags too in a file or keep in html flow.
Ok with you it's not clean in my proposition...

Reply:
@fdonnet Hmmm, I get your point. IMHO, I would put all meta into one file, you know, so we have the cleanest HTML flow possible.

What does the bosses think? @NeOMakinG 🚀

Reply:
I feel like you're right, but this file is getting way too big, what about splitting json lds into multiple files inside a folder, and include it inside a main file ? 

@PierreRambaud What's your advice on this PR ? I like it as I always prefer JSON LD separated from the markup, instead of polluting markup

Reply:
maybe something like that =>

seo-organization-jsonld.tpl
seo-webpage-jsonld.tpl
seo-indexpage-jsonld.tpl
seo-productpage-jsonld.tpl
seo-listpage-jsonld.tpl
seo-breadcrumb-jsonld.tpl

seo-metatags.tpl

all included in seo-jsonld-metatags.tpl

and this seo-jsonld-metatags.tpl included in head.tpl ?

Reply:
We discussed about it with @PierreRambaud 

One wrong thing with the workflow you did, is that on every pages, you'll execute conditions which we know are not needed on a particular page.

The good workflow would be : Split meta datas into multiple files inside a jsonld folder, and include each parts needed inside each template, avoiding the "all in one" file, that would permit to avoid unneeded conditions

wdyt @fdonnet @Hlavtox ?

Reply:
hum, for me it's better to have the jsonld in the <head> tag compared to in <body>... You open your src page and see exactly what seo data you have at the top. I prefer to know that all the meta are included in head.tpl (template central point) to not need to remember in which template I need to go (this one has json, this one not etc)... personnal pref I think ?

Reply:
Actually you can include these in the head, but from each respective template

That wouldn't be too hard to override from child themes as you've to override jsonld templates, it would even be more maintainable as if someone override the product json ld template for example, it won't break the update from PS on others template

Reply:
oh ok, sry but mega presta noob here, do you have a link I can check to know how to change header for a respective template file. I was thinking it' was only via (page condition in head.tpl) :)

Reply:
I would keep it in one file, the flow is steered pretty soon, only `if page = product` and so on.

But, if needed, we can separate it to product, productlist and index. The rest of microdata is common to all pages.

Reply:
Hi everyone. The reason why I'm against having one file with a lot of condition is because of this:

```
{if $page.page_name == 'category' || $page.page_name == 'prices-drop' || $page.page_name == 'new-products' || $page.page_name == 'best-sales'}
```

You assume it only works for these pages, but what if I create a new page named "super-mega-discount-prices", I can't include `themes/classic/templates/_partials/microdata-jsonld.tpl` because it becomes useless nothing will be displayed and I will have to redo the job, or override this template, and no longer follow Core changes.
But if files are separated I can now include `themes/classic/templates/_partials/microdata/product-list-jsonld.tpl`. So it's easier to use :)

Reply:
Ok, so I will explode the files and integrate where needed, but can you give me an example to do it properly 


```
{block name='content'}
  <section id="main">

    {block name='product_list_header'}
      <h2 id="js-product-list-header" class="h2">{$listing.label}</h2>
    {/block}
```
here the top of product-list.tpl . If i put : {include file="_partials/microdata-product-list-jsonld.tpl"} in content block or above... it will be in html body tag ? Do I have a way to extend the header (head html tag) without erase the existing content from this lvl of templates ?

Reply:
![image](https://user-images.githubusercontent.com/14963751/106467103-7e34d700-649c-11eb-83c1-6e526499506b.png)
Maybe this hook will do it ?

Or maybe we can add a new block only for JSON LD, to avoid side effects

Reply:
See `classic/templates/_partials/head.tpl`, maybe a block there could do the job :)

Reply:
discussion continues below